compiler/src/dmd/cparse.d

```diff
@@ -4880,6 +4886,31 @@ final class CParser(AST) : Parser!AST
         return stc;
     }
 
+    /***********************
+     * Add attributes from Specifier to function
+     * Params:
+     *  fd = function to apply them to
+     *  specifier = specifiers
+     */
+    void specifiersToFuncDeclaration(AST.FuncDeclaration fd, const ref Specifier specifier)
+    {
+        fd.isNaked = specifier.naked;
+        fd.dllImport = specifier.dllimport;
+        fd.dllExport = specifier.dllexport;
+    }
+
+    /***********************
+     * Add attributes from Specifier to variable
+     * Params:
+     *  vd = function to apply them to
+     *  specifier = specifiers
+     */
+    void specifiersToVarDeclaration(AST.VarDeclaration vd, const ref Specifier specifier)
+    {
+        vd.dllImport = specifier.dllimport;
+        vd.dllExport = specifier.dllexport;
+    }
```

Comment:
Just wrap it in a VisibilityDeclaration, no reason to add needless duplicate information.

Reply:
VisibilityDeclaration doesn't deal with dllimport/dllexport. If it was added, then it still would have to be propagated from the VisibilityDeclaration to the var/func declaration during the semantic pass. The duplication will still be there. I don't think this is a win.

The point to the VisibilityDeclaration was to enable the block scope of such attributes. This is irrelevant for dllimport/dllexport, which never apply to anything but the specific declaration the are associated with.

You could argue that it should set the Visibility export bit, but that has slightly different semantics than dllexport/dllimport, i.e. there is some impedance mismatch.